apps/settings/js/panels/call/call_forwarding.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * Make sure all operations would be executed one by one.
+ *
+ * @module call/call_forwarding
+ */
+define(function(require) {
+  'use strict';
+
+  var CallSettingsTaskScheduler =
+    require('panels/call/call_settings_task_scheduler');
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+  var CallUtils = require('panels/call/call_utils');
+  var CallConstant = require('panels/call/call_constant');
+  var mozSettings = window.navigator.mozSettings;
+
+  /**
+   * @class CallForwarding
+   * @returns {CallForwarding}
+   */
+  var CallForwarding = Module.create(function CallForwarding(cardIndex) {
+    this.super(Observable).call(this);
+
+    this._cardIndex = cardIndex;
+    this._cfReasonStates = [0, 0, 0, 0];
+  }).extend(Observable);
+
+  Observable.defineObservableProperty(CallForwarding.prototype, 'state', {
+    value: 'normal'
+  });
+
+  // This will create following obserable properties :
+  // 
+  // unConditionalNumber
+  // unConditionalEnabled
+  // mobileBusyNumber
+  // mobileBusyEnabled
+  // noReplyNumber
+  // noReplyEnabled
+  // notReachableNumber
+  // notReachableEnabled
+  // 
+  Object.keys(CallConstant.CALL_FORWARD_REASON_MAPPING).forEach(
+    (cfOptionName) => {
+      Observable.defineObservableProperty(CallForwarding.prototype,
+        cfOptionName + 'Number', {
+          value: null
```

Comment:
Note that for read-only properties we can manipulate the value by adding an underline as the prefix.